Lib/ipaddress.py

```diff
@@ -1292,7 +1292,7 @@ def __init__(self, address):
             return
 
         # Constructing from a packed address
-        if isinstance(address, bytes):
+        if isinstance(address, bytes) or isinstance(address, bytearray):
             self._check_packed_address(address, 4)
             self._ip = int.from_bytes(address)  # big endian
```

Comment:
as a note, the `int.from_bytes` here will still `memcpy` to go from `bytearray` -> `bytes` internally (gh-132108). So this is just moving the need to copy at the moment, not eliminating entirely.